lib/reporters.js

```diff
@@ -57,14 +57,34 @@ DataDogReporter.prototype.report = function(series, onSuccess, onError) {
 
     if (debug.enabled) {
         // Only call stringify when debugging.
-        debug('Calling add_metrics with %s', JSON.stringify(series));
+        debug('Calling report with %s', JSON.stringify(series));
     }
 
-    callbackify(() => metricsApi.submitMetrics({
-        body: {
-            series
+    // Distributions must be submitted via a different method than other
+    // metrics, so split them up.
+    const metrics = [];
+    const distributions = [];
+    for (const metric of series) {
+        if (metric.type === 'distribution') {
+            distributions.push(metric);
+        } else {
+            metrics.push(metric);
         }
-    }))(callback);
+    }
+
+    let submissions = [];
+    if (metrics.length) {
+        submissions.push(metricsApi.submitMetrics({
+            body: { series: metrics }
+        }));
+    }
+    if (distributions.length) {
+        submissions.push(metricsApi.submitDistributionPoints({
+            body: { series: distributions }
+        }));
+    }
+    
+    callbackify(() => Promise.all(submissions))(callback);
```

Comment:
I noticed that this doesn’t seem to add a lot, and the callback is implemented pretty far away from this. I’m thinking it might be nicer to do this instead, but didn’t want to change the overall structure for no reason:

```js
Promise.all(submissions)
    .then(() => {
        debug('sent metrics successfully');
        if (typeof onSuccess === 'function') {
            onSuccess();
        }
    })
    .catch((err, res) => {
        debug('ERROR: failed to send metrics %s (err=%s)', res, err);
        if (typeof onError === 'function') {
            onError(err, res);
        }
    });
```

Reply:
That seems reasonable to me; feel free to implement in a follow-up PR for further discussion.

Reply:
Covered this in #78.